tests/unittests/juju/juju_cluster_tests.py

```diff
@@ -291,4 +292,29 @@ def test_subordinates(self, mock_exec_primary_cmd):
         )
 
 
+class IndexTest(unittest.TestCase):
+
+    def test_subordinate_parent_miss_units(self):
+        """Fix if subordinate's parent is missing units."""
+        model = "sos"
+        index = _get_index(model_name=model)
+
+        juju_status = json.loads(get_juju_output(model=model))
+        juju_status["applications"]["ubuntu"].pop("units")
+
+        index.add_principals(juju_status)
+        index.add_subordinates(juju_status)
```

Comment:
Good to have tests for it. Just my first thought was "and what do we technically test here, isn't an `assert*` missing here?". Maybe a comment like "ensure these commands won't fail even when no `units` are present" would be helpfull.

(the same applies to the 2nd method)

Reply:
Ack